### PR TITLE
Initial TDD Finish

### PR DESCRIPTION
**Pull request by zyelcrier**
- three helper functions to make the main function more readable
-first test for edge cases, then runs program

### CODE REVIEW

**Comment by zyelcrier** [comment]
Potential idea is to take out initial if else because those cases are covered in the helper functions all tests pass without them. Thoughts? (Initial if else in reference to lines 33 and 66)

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 1]
```diff
@@ -15,9 +15,84 @@
...
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    Collection<TimeRange> result = new ArrayList<TimeRange>(); 
+
+    long meetingRequestDuration = request.getDuration();
+    Collection<Event> eventsWithRequestedAttendees = getEventsWithRequestedAttendees(events, request.getAttendees());
+    Collection<TimeRange> availabeRanges = new ArrayList<>();
+
+    if(events.size()==0 && meetingRequestDuration<=TimeRange.WHOLE_DAY.end()){
```
Comment (rayliu0317) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: There should be spaces between if and (. And it should be applied to the rest of the file also.

**Review by rayliu0317 — COMMENTED** [review]
I think this works and is fine for an initial attempt. We can go over the optimal algorithm in our 1:1 on Monday if you want.

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java (hunk 1 above)
Comment (zyelcrier) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I believed i fixed the space you were referring to. Sure we can go through that as well in our 1:1 but do you think that enough time to explain complexity. I also have to do the second part of the week 5 project. @rayliu0317

**Comment by zyelcrier** [comment]
Review Changes:
-Added proper descriptions to the five tests I made 
-changed variable name on collection of time ranges to be more specific
@rayliu0317

File: walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java [hunk 2]
```diff
@@ -30,11 +30,11 @@
...
 
-    Collection<TimeRange> availabeRangesForRequestedAttendees = findAllAvailableTimeRanges(eventsWithRequestedAttendees);
-    Collection<TimeRange> result01 = findViableTimeRangesForRequest(availabeRangesForRequestedAttendees, meetingRequestDuration);
+    Collection<TimeRange> availabeTimeRangesForRequestedAttendees = findAllAvailableTimeRanges(eventsWithRequestedAttendees);
+    Collection<TimeRange> result01 = findViableTimeRangesForRequest(availabeTimeRangesForRequestedAttendees, meetingRequestDuration);
 
-    Collection<TimeRange> availabeRangesForAllAttendees = findAllAvailableTimeRanges(eventsWithAllAttendees);
-    Collection<TimeRange> result02 = findViableTimeRangesForRequest(availabeRangesForAllAttendees, meetingRequestDuration);
+    Collection<TimeRange> availabeRangesForRequestedAndOptionalAttendees = findAllAvailableTimeRanges(eventsWithAllAttendees);
+    Collection<TimeRange> result02 = findViableTimeRangesForRequest(availabeRangesForRequestedAndOptionalAttendees, meetingRequestDuration);
```
Comment (rayliu0317) [inline walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: Still not clear to me what result01 and result02 are supposed to represent.

Comment (zyelcrier) [reply walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java]: I changed those to be specific  I didnt see anything else that I thought was vague or un-descriptive. Does the new change work ? @rayliu0317
